src/codewhisperer/vue/backend.ts

```diff
@@ -0,0 +1,155 @@
+ */
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+import * as os from 'os'
+import * as vscode from 'vscode'
+import * as fs from 'fs'
+import * as path from 'path'
+import { VueWebview } from '../../webviews/main'
+import { isCloud9 } from '../../shared/extensionUtilities'
+import globals from '../../shared/extensionGlobals'
+import { telemetry, CodewhispererLanguage, CodewhispererGettingStartedTask } from '../../shared/telemetry/telemetry'
+
+export class CodeWhispererWebview extends VueWebview {
+    public readonly id = 'CodeWhispererWebview'
+    public readonly source = 'src/codewhisperer/vue/index.js'
+
+    public constructor(private readonly start: string) {
+        super()
+    }
+    //This function is called when the extension is activated to check whether is it the first time the user is using the extension or not
+    public async showAtStartUp(): Promise<string | void> {
+        return this.start
+    }
+
+    private isFileSaved: boolean = false
+    private getLocalFilePath(fileName: string): string {
+        //This will store the files in the global storage path of VSCode
+        return path.join(globals.context.globalStoragePath, fileName)
+    }
+
+    // This function opens Python/JavaScript/C# file in the editor.
+    async openFile(name: string[]): Promise<void> {
+        const fileName = name[0]
+        const fileContent = name[1]
+
+        const localFilePath = this.getLocalFilePath(fileName)
+
+        if (fs.existsSync(localFilePath) && this.isFileSaved) {
+            const fileUri = vscode.Uri.file(localFilePath)
+            vscode.workspace.openTextDocument(fileUri).then(doc => {
+                vscode.window.showTextDocument(doc, vscode.ViewColumn.Active).then(editor => {
+                    const endOfDocument = new vscode.Position(
+                        doc.lineCount - 1,
+                        doc.lineAt(doc.lineCount - 1).text.length
+                    )
+                    editor.selection = new vscode.Selection(endOfDocument, endOfDocument)
+                })
+            })
+        } else {
+            this.saveFileLocally(localFilePath, fileContent)
+        }
+    }
+
+    // This function saves and open the file in the editor.
+    private async saveFileLocally(localFilePath: string, fileContent: string): Promise<void> {
+        await fs.promises.writeFile(localFilePath, fileContent)
+        this.isFileSaved = true
+        vscode.workspace.openTextDocument(localFilePath).then(doc => {
+            vscode.window.showTextDocument(doc, vscode.ViewColumn.Active).then(editor => {
+                const endOfDocument = new vscode.Position(doc.lineCount - 1, doc.lineAt(doc.lineCount - 1).text.length)
+                editor.selection = new vscode.Selection(endOfDocument, endOfDocument)
+            })
+        })
+    }
+
+    //This function returns the OS type of the machine used in Shortcuts and Generate Suggestion Sections
+    public getOSType() {
```

Comment:
Linux is one of the supported platforms

Reply:
As per implementation apart from **Mac** everything is shown as same Key

For Mac: Option+C
Rest : Alt+C

```
 <div class="generateSuggestionTabMachine">
                                    <div class="col2" v-if="osState === 'Mac'">Option</div>
                                    <div class="col2" v-else>Alt</div>
                                    <div class="generateSuggestionTabMachineText">+</div>
                                    <div class="col2">C</div>
                                </div>
```